datafusion/core/src/datasource/listing/url.rs

```diff
@@ -131,6 +131,10 @@ impl ListingTableUrl {
                     if is_directory {
                         fs::create_dir_all(path)?;
                     } else {
+                        // ensure parent directory exists
```

Comment:
Without this I can't write into `/dir/table_name/1.parquet` if `/dir/table_name` doesn't exist already

This is not enabled by deafult, it only affects when the user has specified `SINGLE_FILE_OUTPUT` explicitly

Reply:
Aah, I still need to fix this, this is a weird hold over from append

Reply:
What needs to be fixed? I would like to write up a catalog

Reply:
There is no reason you should need to create these files ahead of time anymore, it was a consequence of the head request being performed to accommodate the append semantics - I can add it to my list to clean it up

Reply:
As in the object_store `LocalFile::put` implementation should automatically create the paths?